src/config/mod.rs

```diff
@@ -466,15 +515,47 @@ impl Display for Resource {
 
 #[derive(Deserialize, Serialize, Debug)]
 #[serde(deny_unknown_fields)]
-pub struct TestDefinition {
+pub struct TestDefinition<Id> {
```

Comment:
It feels strange to make this generic. It seems like it's mostly to allow for deserialization with `String` keys. As mentioned above, maybe all this can go away if we deserialize strings directly into component IDs?